### PR TITLE
shared/vpick: some minor cleanups

### PR DESCRIPTION
**Pull request by YHNdnzj**
Addresses https://github.com/systemd/systemd/pull/32142#discussion_r1572397839

### CODE REVIEW

**Review by bluca — APPROVED** [review]

File: src/shared/vpick.c [hunk 1]
```diff
@@ -185,11 +185,9 @@ static int pin_choice(
...
                 return log_oom_debug();
 
-        if (filter->version) {
-                result.version = strdup(filter->version);
-                if (!result.version)
-                        return log_oom_debug();
-        }
+        r = strdup_to(&result.version, filter->version);
+        if (r < 0)
+                return r;
```
Comment (yuwata) [inline src/shared/vpick.c:188]: `log_oom_debug()` ?

File: src/shared/vpick.c [hunk 2]
```diff
@@ -85,13 +87,9 @@ static int format_fname(
                         return -ENOMEM;
         }
 
-        if (!strv_isempty(filter->suffix)) {
-                if (strv_length(filter->suffix) > 1)
-                        return -ENOEXEC;
-
+        if (!strv_isempty(filter->suffix))
                 if (!strextend(&fn, filter->suffix[0]))
```
Comment (yuwata) [inline src/shared/vpick.c:91]: Not a matter, but you can merge the two `if` conditions.

**Review by yuwata — APPROVED** [review]
LGTM. Some minor comments.

File: src/shared/vpick.c (hunk 1 above)
Comment (YHNdnzj) [reply src/shared/vpick.c:188]: Hmm, so I'm hesitant about this. Generally in library-like codes it's better to propagate the original error, but that means log_oom_debug can't be used...

File: src/shared/vpick.c (hunk 2 above)
Comment (YHNdnzj) [reply src/shared/vpick.c:91]: (I prefer the current form better.)

File: src/shared/vpick.c (hunk 1 above)
Comment (yuwata) [reply src/shared/vpick.c:188]: Yeah, but it only returns -ENOMEM on error... There are so many places we use log_oom() or log_oom_debug() such simple allocation failure with `r == -ENOMEM`.

File: src/shared/vpick.c (hunk 2 above)
Comment (yuwata) [reply src/shared/vpick.c:91]: OK. No problem. Feel free to ignore the suggestion.

File: src/shared/vpick.c (hunk 1 above)
Comment (YHNdnzj) [reply src/shared/vpick.c:188]: Hmm, but this is library-like code? We only do log_oom() directly when the function returns NULL on failure or in user-facing executables?

Comment (poettering) [reply src/shared/vpick.c:188]: log_oom_debug() is not the same as log_oom()

**Comment by poettering** [comment]
lgtm

File: src/shared/vpick.c (hunk 1 above)
Comment (YHNdnzj) [reply src/shared/vpick.c:188]: I know, but the point here is that we should propagate `r`. If log_oom/_debug is used we synthesize `ENOMEM`... And we don't do that in library code

**Comment by YHNdnzj** [comment]
I'll just go ahead and merge this. I really think we should propagate the original error, and this is pretty trivial.

**Pull request merged**